cis194/week2/wei-ning/src/Exercise1.hs

```diff
@@ -0,0 +1,42 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module Exercise1
+  ( parseMessage
+  , parseMessages
+  ) where
+
+import qualified Log
+import           Text.Parsec
+import           Text.Parsec.String
+
+parseMessages :: String -> [Log.LogMessage]
+parseMessages s =
+  case parse (parseLogMessage `endBy` newline) [] s of
+    Right ls -> ls
+    Left _   -> []
+
+parseMessage :: String -> Log.LogMessage
+parseMessage s =
+  case parse parseLogMessage [] s of
+    Right l -> l
+    Left _  -> Log.Unknown s
+
+parseLogMessage :: Parser Log.LogMessage
+parseLogMessage = do
+  typeToken <- parseTypeToken
+  many1 (char ' ')
+  timeStamp <- parseTimeStamp
+  many (char ' ')
+  message <- many (noneOf ['\n'])
+  return (Log.LogMessage typeToken timeStamp message)
+
+parseTypeToken :: Parser Log.MessageType
+parseTypeToken =
+      (char 'I' >> return Log.Info)
```

Comment:
Nowadays, we use `pure` instead of `return` in Haskell since we have Applicative as functional constraint on Monad typeclass now.

Reply:
👍